lacework/resource_lacework_integration_azure_agentless_scanning.go

```diff
@@ -0,0 +1,407 @@
+			TenantId:                  d.Get("tenant_id").(string),
+			ScanFrequency:             d.Get("scan_frequency").(int),
+			ScanContainers:            d.Get("scan_containers").(bool),
+			ScanHostVulnerabilities:   d.Get("scan_host_vulnerabilities").(bool),
+			ScanMultiVolume:           d.Get("scan_multi_volume").(bool),
+			ScanStoppedInstances:      d.Get("scan_stopped_instances").(bool),
+			QueryText:                 d.Get("query_text").(string),
+			SubscriptionsList:         strings.Join(castAttributeToStringSlice(d, "subscriptions_list"), ", "),
+		},
+	)
+
+	if !d.Get("enabled").(bool) {
+		data.Enabled = 0
+	}
+
+	return retry.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *retry.RetryError {
+		retries--
+		log.Printf("[INFO] Creating %s integration\n", api.AzureSidekickCloudAccount.String())
+		log.Printf("[INFO] Creating %v integration\n", data)
+		response, err := lacework.V2.CloudAccounts.CreateAzureSidekick(data)
+
+		if err != nil {
+			if retries <= 0 {
+				return retry.NonRetryableError(
+					fmt.Errorf("Error creating %s integration: %s",
+						api.AzureSidekickCloudAccount.String(), err,
+					))
+			}
+			log.Printf(
+				"[INFO] Unable to create %s integration. (retrying %d more time(s))\n%s\n",
+				api.AzureSidekickCloudAccount.String(), retries, err,
+			)
+			return retry.RetryableError(fmt.Errorf(
+				"Unable to create %s integration (retrying %d more time(s))",
+				api.AzureSidekickCloudAccount.String(), retries,
+			))
+		}
+
+		integration := response.Data
+		d.SetId(integration.IntgGuid)
+		d.Set("name", integration.Name)
+		d.Set("intg_guid", integration.IntgGuid)
+		d.Set("enabled", integration.Enabled == 1)
+		d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
+		d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
+		d.Set("server_token", integration.ServerToken)
+		d.Set("uri", integration.Uri)
+
+		log.Printf("[INFO] Created %s integration with guid: %v\n",
+			api.AzureSidekickCloudAccount.String(), integration.IntgGuid)
+		return nil
+	})
+}
+
+func resourceLaceworkIntegrationAzureAgentlessScanningRead(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Reading %s integration with guid: %v\n",
+		api.AzureSidekickCloudAccount.String(), d.Id())
+	response, err := lacework.V2.CloudAccounts.GetAzureSidekick(d.Id())
+	if err != nil {
+		return resourceNotFound(d, err)
+	}
+
+	integration := response.Data
+	if integration.IntgGuid == d.Id() {
+		d.Set("name", integration.Name)
+		d.Set("intg_guid", integration.IntgGuid)
+		d.Set("enabled", integration.Enabled == 1)
+		d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
+		d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
+		d.Set("integration_level", integration.Type)
+
+		creds := make(map[string]string)
+		creds["client_id"] = integration.Data.Credentials.ClientId
+		d.Set("credentials", []map[string]string{creds})
```

Comment:
do we need to set client secret?

Reply:
good question. I copied it from the GCP flow. Apparently it was intentionally removed (https://github.com/lacework/terraform-provider-lacework/pull/531) so I'm just going to do the same here.